### PR TITLE
#5186: UTWH neighbourhood distance >2 support

### PR DESCRIPTION
**Pull request by nsmithtt**
Config data has been made completely generic such that cores can address
any other core, i.e. any neighborhood distance.  There is now a single
UTWH kernel that is parameterized via compile time args to specify which
of, padding, local, remote data each risc DM core should perform.

Padding config data is still special cased since its access pattern is
different (single source row, always local).

Perf is better for almost every case in resnet.

### CODE REVIEW

**Comment by nsmithtt** [comment]
@mywoodstock / @tt-nshanker, I have both the new and old paths supported!  To simplify the diff process I figured I'd commit the additions first, and then remove the old path as a separate change.  LMK if you want me to do otherwise.

@tt-nshanker, also if you don't mind taking a look to see if I setup the program caching correctly.

**Review by arakhmati — APPROVED** [review]

File: tests/tt_eager/python_api_testing/unit_testing/test_optimized_conv_v2.py [hunk 1]
```diff
@@ -201,3 +200,124 @@ def test_optimized_conv_v2(
     print("Passing=", passing_pcc)
     print("Info=", info)
     assert passing_pcc
+
+
+def test_simple(
+    use_program_cache,
+    device,
+):
```
Comment (nsmithtt) [inline tests/tt_eager/python_api_testing/unit_testing/test_optimized_conv_v2.py:208]: I can remove we you want.  I found this test useful for just one-off debugging, but I can keep it local.

**Comment by nsmithtt** [comment]
Follow on removal of fixed neighborhood path: https://github.com/tenstorrent-metal/tt-metal/pull/5650

File: tests/tt_eager/python_api_testing/unit_testing/test_optimized_conv_v2.py (hunk 1 above)
Comment (nsmithtt) [reply tests/tt_eager/python_api_testing/unit_testing/test_optimized_conv_v2.py]: I ended up removing it since I realized we wanted to deprecate this test

**Comment by nsmithtt** [comment]
Closing, PR moved here https://github.com/tenstorrent-metal/tt-metal/pull/5650

**Pull request closed without merging**